### PR TITLE
[MDS-5477] ESUP Field Validation

### PR DESCRIPTION
**Pull request by isuru-aot**
## Objective 

[MDS-5477](https://bcmines.atlassian.net/browse/MDS-5477)

_Why are you making this change? Provide a short explanation and/or screenshots_

Screenshot 2023-09-18 150142

Screenshot 2023-09-18 155030

### CODE REVIEW

**Review by asinn134 — DISMISSED** [review]

File: services/core-web/src/components/Forms/ExplosivesPermit/MagazineForm.js [hunk 1]
```diff
@@ -134,7 +134,7 @@ export class MagazineForm extends Component {
                 label="Longitude *"
                 id={`${field}longitude`}
                 name={`${field}longitude`}
-                validate={[number, maxLength(12), lon, required]}
+                validate={[number, maxLength(12), lonNegative, required]}
```
Comment (simensma-fresh) [inline services/core-web/src/components/Forms/ExplosivesPermit/MagazineForm.js]: Think this needs to have both `lonNegative` and `lon` to make sure it is both a valid longitude and is negative

File: services/minespace-web/common/utils/Validate.js [hunk 2]
```diff
@@ -108,6 +108,11 @@ export const exactLength = memoize((min) => (value) =>
 export const number = (value) =>
   value && Number.isNaN(Number(value)) ? "Input must be a number" : undefined;
 
+export const positiveNumber = (value) =>
```
Comment (simensma-fresh) [inline services/minespace-web/common/utils/Validate.js:111]: Would be awesome if we could add a couple of tests for this one, the other validations have tests defined in `src/tests/utils/Validate.spec.js`

**Review by simensma-fresh — CHANGES_REQUESTED** [review]

File: services/minespace-web/common/utils/Validate.js (hunk 2 above)
Comment (isuru-aot) [reply services/minespace-web/common/utils/Validate.js:111]: @simensma-fresh  A test added for `positiveNumber`.

**Review by henryoforeh-dev — DISMISSED** [review]
Nice work!

**Review by matbusby-fw — DISMISSED** [review]
Awesome!  Looks great!

File: services/core-web/src/components/Forms/ExplosivesPermit/MagazineForm.js (hunk 1 above)
Comment (isuru-aot) [reply services/core-web/src/components/Forms/ExplosivesPermit/MagazineForm.js]: @simensma-fresh this is updated accordingly.

**Review by henryoforeh-dev — APPROVED** [review]

**Review by asinn134 — APPROVED** [review]

**Review by simensma-fresh — APPROVED** [review]

**Pull request merged**